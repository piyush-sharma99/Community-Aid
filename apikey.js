/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
