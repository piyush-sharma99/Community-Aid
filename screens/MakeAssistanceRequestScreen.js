import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, TextInput, ScrollView, Button, TouchableOpacity, Image} from 'react-native';
import {Card} from 'react-native-paper';
import * as firebase from 'firebase';
import * as Location from 'expo-location';
import makeRequest from '../functions/makeRequest';

const MakeAssistanceRequestScreen = props => {

  const db = firebase.firestore();
  const [location, setLocation] = useState(null);
  const [errorMsg, setErrorMsg] = useState(null);
  const requestID = Math.random().toString(36).slice(2); //Incrementing a random key
  const [date, setDate] = useState('');
  const [status] = useState('Unassigned');
  const [address, setAddress] = useState('');
  const [area, setArea] = useState('');
  const [requestType, setRequestType] = useState('');
  const [requestDescription, setRequestDescription] = useState('');
  var user = firebase.auth().currentUser;

//async function that pulls users latitude and longitude
  useEffect(() => {
    (async () => {
      let { status } = await Location.requestPermissionsAsync();
      if (status !== 'granted') {
        setErrorMsg('Permission to access lo cation was denied');
        return;
      }

      let location = await Location.getCurrentPositionAsync({});
      setLocation(location);
    })();
  }, []);

  let text = 'Waiting..';
  let latitude = -9.062691;
  let longitude = 53.270962;
  if (errorMsg) {
    text = errorMsg;
  } else if (location) {
    text = JSON.stringify(location);
    longitude = location.coords.longitude;
    latitude = location.coords.latitude;
    
  }

  resetFields = () => {
    this.textInputOne.clear();
    this.textInputTwo.clear();
    this.textInputThree.clear();
    this.textInputFour.clear();
    this.textInputFive.clear();
        
}


    return(

      <ScrollView style={styles.scroll}>
        <View style={styles.screen} >

<Text style={styles.text}>Fill the form below:</Text>
         <Card style={styles.cardView}>

         <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Date" 
            placeholderTextColor="#003f5c"
            onChangeText={(date) => setDate(date)}
            ref={input => { this.textInputOne = input }}/>
        </View>

      
         <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Address..." 
            placeholderTextColor="#003f5c"
            onChangeText={(address) => setAddress(address)}
            ref={input => { this.textInputTwo = input }}/>
        </View>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Area..." 
            placeholderTextColor="#003f5c"
            onChangeText={(area) => setArea(area)}
            ref={input => { this.textInputThree = input }}/>
        </View>

        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Request Type..." 
            placeholderTextColor="#003f5c"
            onChangeText={(requestType) => setRequestType(requestType)}
            ref={input => { this.textInputFour = input }}/>
        </View>

        <View style={styles.inputViewBig} >
          <TextInput  
            style={styles.inputText}
            placeholder="Request Description..." 
            placeholderTextColor="#003f5c"
            multiline={true}
            onChangeText={(requestDescription) => setRequestDescription(requestDescription)}
            ref={input => { this.textInputFive = input }}/>
        </View>

        </Card>

        <View>
        <TouchableOpacity style={styles.Btn} onPressIn = {() => makeRequest(user, requestID, db, date, status, address, area, requestType, requestDescription, longitude, latitude)}
        onPress = {resetFields}>
          <Text style={styles.Text}>Submit</Text>
        </TouchableOpacity>
        </View>

        

        </View>
        </ScrollView>
    );
};

MakeAssistanceRequestScreen.navigationOptions = {
headerTitle: 'Make Assistance Request',
headerTitleStyle: { alignSelf: 'center' },
headerStyle: {
backgroundColor: '#2E86C1'
},
headerTintColor:"white"
};


const styles = StyleSheet.create({

  scroll: {
    width:'100%',
    height:'100%',
    backgroundColor: '#2E86C1',
},
  text: {

    padding:10,
    marginBottom: 30,
    marginRight:40,
    marginLeft:40,
    color:"white",
    fontSize:30,
    textAlign: 'center',
    marginTop:30,
    borderWidth: 3,
    borderColor: '#fb5b5a',
    borderRadius:15,

    },
    screen: {
        width:'100%',
        height:'100%',
        flex: 1,
        backgroundColor: '#2E86C1',
        alignItems: 'center',
        justifyContent: 'center'
    },
      inputView:{
        width:300,
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:50,
        marginTop:25,
        marginBottom:15,
        justifyContent:"center",
        padding:20,
        elevation: 10,
        borderWidth: 4,
        borderColor: "#2E86C1"
      },
      inputViewBig:{
        width:300,
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:130,
        marginTop:25,
        marginBottom:15,
        justifyContent:"center",
        padding:20,
        elevation: 10,
        borderWidth: 4,
        borderColor: "#2E86C1"
      },

      inputText:{
        height:60,
        color:"black"
      },
      inputTextTwo:{
        height:60,
        color:"black"
      },

      Text:{
        color:"white",
        fontSize:20
      },

      
      Btn:{
        width:300,
        backgroundColor:"#fb5b5a",
        borderRadius:15,
        height:50,
        alignItems:"center",
        justifyContent:'center',
        marginTop:30,
        borderWidth:2,
        borderColor: "white"
      },
      cardView: {
        justifyContent: 'center',
        width:'90%',
        height:'77%',
        backgroundColor:"#fb5b5a",
        shadowColor: 'black',
        borderRadius:25,
        shadowOpacity: 1,
        elevation: 10,
        alignItems:"center",
        shadowOffset: {
            width: 3,
            height: 3
        },
    },
      
      
});

export default MakeAssistanceRequestScreen;