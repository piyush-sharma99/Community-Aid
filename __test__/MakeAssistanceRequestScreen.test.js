import React from "react";
import MakeAssistanceRequestScreen from "../screens/MakeAssistanceRequestScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("MakeAssistanceRequest renders correctly", () => {
  const tree = renderer.create(<MakeAssistanceRequestScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
