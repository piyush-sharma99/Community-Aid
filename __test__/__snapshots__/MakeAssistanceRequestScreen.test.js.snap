// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeAssistanceRequest renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "#fb5b5a",
              "borderRadius": 15,
              "borderWidth": 3,
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "marginBottom": 30,
              "marginLeft": 40,
              "marginRight": 40,
              "marginTop": 30,
              "padding": 10,
              "textAlign": "center",
            }
          }
        >
          Fill the form below:
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fb5b5a",
              "borderColor": "white",
              "borderRadius": 25,
              "borderWidth": 3,
              "elevation": 10,
              "height": "77%",
              "justifyContent": "center",
              "marginTop": 30,
              "padding": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 10,
              "width": "90%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              index={0}
              siblings={
                Array [
                  "View",
                  "View",
                  "View",
                  "View",
                ]
              }
              style={
                Object {
                  "backgroundColor": "#FDFEFE",
                  "borderColor": "#2E86C1",
                  "borderRadius": 25,
                  "borderWidth": 4,
                  "elevation": 10,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 15,
                  "marginTop": 25,
                  "padding": 20,
                  "width": 300,
                }
              }
              total={5}
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Address..."
                placeholderTextColor="#003f5c"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "black",
                    "height": 60,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              index={1}
              siblings={
                Array [
                  "View",
                  "View",
                  "View",
                  "View",
                ]
              }
              style={
                Object {
                  "backgroundColor": "#FDFEFE",
                  "borderColor": "#2E86C1",
                  "borderRadius": 25,
                  "borderWidth": 4,
                  "elevation": 10,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 15,
                  "marginTop": 25,
                  "padding": 20,
                  "width": 300,
                }
              }
              total={5}
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Area..."
                placeholderTextColor="#003f5c"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "black",
                    "height": 60,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              index={2}
              siblings={
                Array [
                  "View",
                  "View",
                  "View",
                  "View",
                ]
              }
              style={
                Object {
                  "backgroundColor": "#FDFEFE",
                  "borderColor": "#2E86C1",
                  "borderRadius": 25,
                  "borderWidth": 4,
                  "elevation": 10,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 15,
                  "marginTop": 25,
                  "padding": 20,
                  "width": 300,
                }
              }
              total={5}
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Request Type..."
                placeholderTextColor="#003f5c"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "black",
                    "height": 60,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              index={3}
              siblings={
                Array [
                  "View",
                  "View",
                  "View",
                  "View",
                ]
              }
              style={
                Object {
                  "backgroundColor": "#FDFEFE",
                  "borderColor": "#2E86C1",
                  "borderRadius": 25,
                  "borderWidth": 4,
                  "elevation": 10,
                  "height": 100,
                  "justifyContent": "center",
                  "marginBottom": 15,
                  "marginTop": 25,
                  "padding": 20,
                  "width": 300,
                }
              }
              total={5}
            >
              <TextInput
                allowFontScaling={true}
                multiline={true}
                onChangeText={[Function]}
                placeholder="Request Description..."
                placeholderTextColor="#003f5c"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "black",
                    "height": 60,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2E86C1",
                  "borderColor": "white",
                  "borderRadius": 15,
                  "borderWidth": 2,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "marginTop": 20,
                  "opacity": 1,
                  "width": 300,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  }
                }
              >
                Submit
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
